src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1000,21 +1000,12 @@ def parse(
             else:
                 pre_nc = ()
                 post_nc = ()
-                if (not segments[0].is_code) and (
```

Comment:
No changes in functionality in this area. Just refactored this code into separate functions for reuse elsewhere.